### PR TITLE
fix(components): [TreeSelect] incorrect label when child not rendered

### PR DESCRIPTION
**Pull request by yujinpan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Fix:

- #10323
- #9749
- #8495

### CODE REVIEW

File: docs/en-US/component/tree-select.md [hunk 1]
```diff
@@ -107,3 +99,9 @@ and please go to the original component to view the documentation.
 | --------------------------------------- | ----------------------------- | ----------------------------------- | ---------------------------------- |
 | [tree](./tree.md#attributes)            | [tree](./tree.md#method)      | [tree](./tree.md#events)            | [tree](./tree.md#slots)            |
 | [select](./select.md#select-attributes) | [select](./select.md#methods) | [select](./select.md#select-events) | [select](./select.md#select-slots) |
+
+### Own Attributes
+
+| Name                                    | Description                                                                                                         | Type  | Accepted Values | Default |
+|-----------------------------------------| ------------------------------------------------------------------------------------------------------------------- | ----- | --------------- | ------- |
+| cacheData<VersionTag version="2.2.3" /> | The cached data of the lazy node, the structure is the same as the data, used to get the label of the unloaded data | array | —               | —       |
```
Comment (xiaoxian521) [inline docs/en-US/component/tree-select.md:107]: use `2.2.26`

File: docs/examples/tree-select/lazy.vue [hunk 2]
```diff
@@ -1,11 +1,23 @@
 <template>
   <el-tree-select v-model="value" lazy :load="load" :props="props" />
+  <el-divider />
+  <VersionTag version="2.2.3" /> show lazy load label:
```
Comment (xiaoxian521) [inline docs/examples/tree-select/lazy.vue:4]: ditto

File: packages/components/tree-select/src/cache-options.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { defineComponent, inject } from 'vue'
+import { selectKey } from '@element-plus/components/select'
+import type { SelectContext } from '@element-plus/components/select'
+import type { PropType } from 'vue'
+
+export type CacheOption = {
```
Comment (xiaoxian521) [inline packages/components/tree-select/src/cache-options.ts:6]: Maybe the following type is better 🤔
```
export type CacheOption = {
  value: string | number | Record<string, string>
  label: string | number
  isDisabled: boolean
}
```

Comment (yujinpan) [reply packages/components/tree-select/src/cache-options.ts:8]: Yes, any type is bad, has been modified to be consistent with el-option instances.

https://github.com/element-plus/element-plus/pull/10716/commits/3fd07a2c9c7e6c4338486b52bb95e9cbcadaf1e2

Comment (yujinpan) [reply packages/components/tree-select/src/cache-options.ts:8]: https://github.com/element-plus/element-plus/pull/10716/commits/52ae3a3fe342c82f9cf7fcf21bd091a4d4f32d61

File: docs/examples/tree-select/lazy.vue (hunk 2 above)
Comment (yujinpan) [reply docs/examples/tree-select/lazy.vue]: https://github.com/element-plus/element-plus/pull/10716/commits/52ae3a3fe342c82f9cf7fcf21bd091a4d4f32d61

File: packages/components/tree-select/src/cache-options.ts (hunk 3 above)
Comment (yujinpan) [reply packages/components/tree-select/src/cache-options.ts:8]: Yes, any type is bad, has been modified to be consistent with el-option instances.

https://github.com/element-plus/element-plus/pull/10716/commits/3fd07a2c9c7e6c4338486b52bb95e9cbcadaf1e2

File: docs/en-US/component/tree-select.md (hunk 1 above)
Comment (yujinpan) [reply docs/en-US/component/tree-select.md]: https://github.com/element-plus/element-plus/commit/52ae3a3fe342c82f9cf7fcf21bd091a4d4f32d61

**Review by xiaoxian521 — APPROVED** [review]

**Pull request merged**